arbor/backends/gpu/mechanism.cpp

```diff
@@ -170,15 +173,15 @@ void mechanism::instantiate(unsigned id,
         std::vector<index_type> mech_ion_index(indices.begin(), indices.end());
 
         // Take reference to derived (generated) mechanism ion index pointer.
-        auto& ion_index_ptr = *ion_index_tbl[i].second;
-        auto index_start = indices_.data()+(i+1)*width_padded_;
-        ion_index_ptr = index_start;
-        memory::copy(make_const_view(mech_ion_index), device_view(index_start, width_));
+        memory::copy(make_const_view(mech_ion_index), device_view(base_ptr, width_));
+        *ion_ptr = base_ptr;
+        base_ptr += width_padded_;
     }
 
     if (mult_in_place_) {
-        memory::copy(make_const_view(pos_data.multiplicity), device_view(indices_.data() + width_padded_, width_));
-        pp->multiplicity_ = indices_.data() + (num_ions_ + 1)*width_padded_;
+        memory::copy(make_const_view(pos_data.multiplicity), device_view(base_ptr, width_));
+        pp->multiplicity_ = base_ptr;
+        base_ptr += width_padded_; // Theoretically redundant, but for consistency
```

Comment:
minor comment: this line is not needed right?

Reply:
Yes, therefore the comment. It's more consistent and also makes it safer to tack on another chunk.

Reply:
I did.